MapboxNavigation/NavigationMapView.swift

```diff
@@ -383,16 +400,53 @@ open class NavigationMapView: MGLMapView {
                     layer.identifier != arrowLayerIdentifier && layer.identifier != arrowSymbolLayerIdentifier && layer.identifier != arrowCasingSymbolLayerIdentifier && layer.identifier != arrowLayerStrokeIdentifier && layer.identifier != waypointCircleIdentifier {
                     style.insertLayer(line, below: layer)
                     style.insertLayer(lineCasing, below: line)
-                    return
+                    break
                 }
             }
         }
+        guard routes.count > 1 else { return }
+        var tmpRoutes = routes
+        tmpRoutes.remove(at: activeRouteIndex)
+        guard let alternateRoute = tmpRoutes.first else { return }
+        
+        self.routes = routes
+        
+        let polyline = MGLPolylineFeature(coordinates: alternateRoute.coordinates!, count: alternateRoute.coordinateCount)
+        polyline.attributes[routeIndexAttribute] = routes.index(of: alternateRoute)
+        let alternatePolyline = MGLShapeCollectionFeature(shapes: [polyline])
+        
+        if let source = style.source(withIdentifier: alternateSourceIdentifier) as? MGLShapeSource {
+            source.shape = alternatePolyline
+        } else {
+            let alternateSource = MGLShapeSource(identifier: alternateSourceIdentifier, shape: alternatePolyline, options: nil)
+            style.addSource(alternateSource)
+            
+            let alternateLayer = routeAlternateStyleLayer(identifier: alternateLayerIdentifier, source: alternateSource)
+            
+            if let layer = style.layer(withIdentifier: routeLayerCasingIdentifier) {
+                style.insertLayer(alternateLayer, below: layer)
+            }
+        }
+    }
+    
+    /**
+     Fired when the user taps a route. If there is an alternate route, and an alternate route is tapped, it is selected.
+     */
+    public func didTapRoute(tap: UITapGestureRecognizer) {
```

Comment:
Why does this method need to be public, given that it’s the action method of a gesture recognizer that’s internal to this class?